raiden/utils/upgrades.py

```diff
@@ -140,10 +144,16 @@ def run(self):
                 with storage.transaction():
                     version_iteration = older_version
                     for upgrade_func in UPGRADES_LIST:
+                        if upgrade_func == upgrade_v18_to_v19:
+                            extra_args = {'web3': self._web3}
+                        else:
+                            extra_args = {}
```

Comment:
please, just change `upgrade_v16_to_v17` and `upgrade_v17_to_v18` to receive `**kwargs`, otherwise adding new dependencies to the upgrade functions will be a pain.

Reply:
Hey @hackaugusto I am not sure what you mean by this comment here. The place in the code you are pointing populates the `extra_args` for the `_run_upgrade_func()` which in turn forwards them to the corresponding upgrade function. At the moment only `upgrade_v18_to_v19` accepts some extra kwargs and in other cases they are empty.

How would adding kwargs to `upgrade_v16_to_v17` and `upgrade_v17_to_v18` make things better?

They don't need any extra args, so giving them an empty `extra_args` works and does nothing. Can you maybe elaborate on what you would like to see changed?

Reply:
I don't want to be in a position where everything I need `web3` in a upgrade function I have to come back here and update this if statement, this also applies to other resources which may required in future transactions, so the idea is to *unconditionally* add `web3` to the `extra_args`.

If `extra_args` always have `web3` in it, `_run_upgrade_func` will forward to the underlying function (which may be ATM any of the three, v16_to_v17, v17_to_v17, or v17_to_v19). `_run_upgrade_func` will in turn unconditionally forward apply `**extra_args` to the function [here](https://github.com/raiden-network/raiden/pull/3550/files#diff-f6569e9ea8f2374edc43e4318c1359f4R63), this means the call to `upgrade_v16_to_v17` [which does not accept **kwargs](https://github.com/raiden-network/raiden/pull/3550/files#diff-9b24031f75a5bc3cb164c9130492b378R52) will fail with a `TypeError: got an unexpected keyword argument`, in order to avoid the error all upgrade function should have `**kwargs` added, this will allow us to add new dependencies for future upgrade functions, unconditionally add them to the `extra_args` dictionary, and have backward compatibility (meaning we don't need to upgrade the existing functions)

Reply:
Aha okay gotcha. That makes sense. Will adjust.